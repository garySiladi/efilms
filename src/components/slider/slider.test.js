// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Slider from './slider';

const dummySliderItems = [
  {
    title: 'Episode 1 Title',
    type: 'Episode 1',
  },
  {
    title: 'Episode 2 Title',
    type: 'Episode 2',
    thumbnail: {
      thumb: {
        url: 'xxx',
      },
    },
  },
  {
    title: 'Some title',
    type: 'Series',
  },
];

const additionalProps = {
  className: 'home-slider',
  sliderTitle: 'Slider title',
};

const sliderDummyProps = [
  {
    ...additionalProps,
    data: dummySliderItems,
    isSelected: true,
    isBeforeSelected: false,
    isHidden: false,
    selectedEpisode: 4,
  },
  {
    ...additionalProps,
    data: dummySliderItems,
    isSelected: false,
    isBeforeSelected: true,
    isHidden: true,
    selectedEpisode: 10,
  },
  {
    ...additionalProps,
    data: dummySliderItems,
    isSelected: false,
    isBeforeSelected: false,
    isHidden: false,
    selectedEpisode: 9,
  },
  {
    ...additionalProps,
    data: dummySliderItems,
    isSelected: false,
    isBeforeSelected: false,
    isHidden: false,
    selectedEpisode: 122,
    isEpisodeExpanded: true,
    isAfterSelected: true,
  },
];

jest.mock('./parts/slider-item', () =>
  jest.fn(() => <div>SliderItem</div>),
);

test('Slider renders correctly', () => {
  sliderDummyProps.forEach((props) => {
    const tree : string = renderer.create(<Slider {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
