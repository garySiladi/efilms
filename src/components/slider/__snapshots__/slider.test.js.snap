// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div
  className="home-slider home-slider--selected"
>
  <div
    className="slider-title"
  >
    Slider title
  </div>
  <div
    className="slider-items"
    style={
      Object {
        "transform": "translate(-320px, 0px)",
      }
    }
  >
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly 2`] = `
<div
  className="home-slider"
>
  <div
    className="slider-title"
  >
    Slider title
  </div>
  <div
    className="slider-items"
    style={
      Object {
        "transform": "translate(0px, 0px)",
      }
    }
  >
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly 3`] = `
<div
  className="home-slider"
>
  <div
    className="slider-title"
  >
    Slider title
  </div>
  <div
    className="slider-items"
    style={
      Object {
        "transform": "translate(0px, 0px)",
      }
    }
  >
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
  </div>
</div>
`;

exports[`Slider renders correctly 4`] = `
<div
  className="home-slider home-slider--after-expanded"
>
  <div
    className="slider-title"
  >
    Slider title
  </div>
  <div
    className="slider-items"
    style={
      Object {
        "transform": "translate(0px, 0px)",
      }
    }
  >
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
    <div>
      SliderItem
    </div>
  </div>
</div>
`;
